core/include/vecmem/edm/impl/host.ipp

```diff
@@ -65,6 +65,22 @@ VECMEM_HOST void host<schema<VARTYPES...>, INTERFACE>::reserve(
                                        std::index_sequence_for<VARTYPES...>{});
 }
 
+template <typename... VARTYPES, template <typename> class INTERFACE>
+VECMEM_HOST void host<schema<VARTYPES...>, INTERFACE>::push_back(
+    const object_type& element) {
+
+    // Make sure that there are some (jagged) vector types in the container.
+    static_assert(
+        std::disjunction_v<type::details::is_vector<VARTYPES>...>,
+        "This function requires at least one (jagged) vector variable.");
+
+    // Add a new element to the container.
+    const size_type index = size();
+    resize(index + 1);
+    // Set the new element.
+    at(index) = element;
```

Comment:
The implementation here is super naive, but again, I thought this could be good enough. But let me know if you guys know of a non-too-verbose way of doing this in a more efficient way.